carapace-server/src/main/java/org/carapaceproxy/server/mapper/requestmatcher/MatchingContext.java

```diff
@@ -0,0 +1,35 @@
+/*
+ Licensed to Diennea S.r.l. under one
+ or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+package org.carapaceproxy.server.mapper.requestmatcher;
+
+/**
+ *
+ * @author paolo.venturi
+ */
+public interface MatchingContext {
+
+    /**
+     *
+     * @param name
+     * @return property value
+     * @throws MatchingException whether the property does not exist.
+     */
+    Object getProperty(String name) throws MatchingException;
```

Comment:
it is better to always return a String, this way is won't be ambiguous and we will save a lot of casts and type checks.
we are returns headers or other textual properties